### ISSUE TITLE
SSL Error

### ISSUE THREAD

**Issue by dsrukmana** [thp/urlwatch#265]
I try to get notified when changes made on this web https://sscn.bkn.go.id but it throw an error, 

first :
ERROR: bkn (https://sscn.bkn.go.id)
---------------------------------------------------------------------------
HTTPSConnectionPool(host='sscn.bkn.go.id', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:833)'),))
---------------------------------------------------------------------------

second : 
urlwatch 2.8, Copyright 2008-2018 Thomas Perl
Website: http://thp.io/2008/urlwatch/
watched 1 URLs in 0 seconds
Failed to load or connect to pushover - are the dependencies installed and configured?
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/urlwatch/reporters.py", line 386, in submit
    service = self.web_service_get()
  File "/usr/lib/python3/dist-packages/urlwatch/reporters.py", line 401, in web_service_get
    app = chump.Application(self.config['app'])
  File "/usr/local/lib/python3.6/dist-packages/chump/__init__.py", line 208, in __init__
    self.token = unicode(token) #: A :py:obj:`string` of the application's API token.
  File "/usr/local/lib/python3.6/dist-packages/chump/__init__.py", line 219, in __setattr__
    raise ValueError('Bad application token: expected string matching [a-zA-Z0-9]{{30}}, got {value!r}'.format(value=value))
ValueError: Bad application token: expected string matching [a-zA-Z0-9]{30}, got 'bkn-monitor'

I use Ubuntu 18.04

**Comment by cfbao** [comment]
The SSL issue seems to be in `requests` or even a lower level library.

As a workaround, you can either disable SSL verification by settings `ssl_no_verify: true` in job config, or use "browser" job after #310 is merged.

Your second error isn't related to SSL. You'd better open a new issue with appropriate title for that.

**Comment by Hitaro** [comment]
Same error:

    ERROR: https://<IP_Address>
    ---------------------------------------------------------------------------
    HTTPSConnectionPool(host='<IP_Address>', port=****): Max retries exceeded with url: <Location> (Caused by SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c...)')))

--

> you can either disable SSL verification by settings ssl_no_verify: true in job config

you mean urlwatch file?

**Comment by cfbao** [comment]
These are actually two different kinds of errors.

@dsrukmana's error has to do with verifying certificate. In this case, because the site doesn't provide the full certificate chain, `requests` cannot verify it. Browsers can open it fine, because browser cache some intermediate CA certificates, whereas `requests` only has root CA certificates.
Disabling certificate verification can bypass this error. It can be fixed by implementing #356, so users can provide a CA bundle that contains the necessary intermediate CA certificate.

@Hitaro's error is with SSL handshake. So disabling certificate checking doesn't help. I'd guess this is the same issue with #361, i.e. `requests` has a very strict (secure) SSL/TLS cipher suite that is incompatible with what the site is using. Fixing it would require a setting to configure SSL/TLS version and/or cipher suites.

**Comment by cfbao** [comment]
@thp I suggest this issue be closed and superseded with #356.
